src/main/java/htsjdk/variant/vcf/AbstractVCFCodec.java

```diff
@@ -719,7 +719,10 @@ private static boolean isVCFStream(final InputStream stream, final String MAGIC_
                         } else if (gtKey.equals(VCFConstants.GENOTYPE_PL_KEY)) {
                             gb.PL(decodeInts(genotypeValues.get(i)));
                         } else if (gtKey.equals(VCFConstants.GENOTYPE_LIKELIHOODS_KEY)) {
-                            gb.PL(GenotypeLikelihoods.fromGLField(genotypeValues.get(i)).getAsPLs());
+                            // Do not overwrite PL with data from GL
+                            if (!gb.make().hasPL()) {
+                                gb.PL(GenotypeLikelihoods.fromGLField(genotypeValues.get(i)).getAsPLs());
```

Comment:
@yfarjoun Thanks for fixing this.  I don't love the call to make() in this if statement.  It would be nice to avoid any extra allocations.  It might be worth adding a getPL method to the builder in order to avoid having to do that.